xds/internal/xdsclient/xdsresource/cluster_resource_type.go

```diff
@@ -46,6 +48,14 @@ type clusterResourceType struct {
 	resourceTypeState
 }
 
+func init() {
+	if internal.ResourceTypeMapForTesting == nil {
+		internal.ResourceTypeMapForTesting = make(map[string]interface{})
+	}
+	internal.ResourceTypeMapForTesting[version.V3ClusterURL] = clusterType
+
+}
+
```

Comment:
Would it be possible to move the registration for all resource types into a single `init()` function in `resource_type.go`? That would also have the added benefit of not having to perform this `nil` check, instead you can initialize the map right there will all four entries.

Reply:
nice! making this update too